### ISSUE TITLE
Enable Framework Selection During Helm Installation

### ISSUE THREAD

**Issue by AdonisOkpidi** [kubescape/helm-charts#566]
## Overview
Currently, all frameworks are scanned by default, and there is no existing mechanism to control this behaviour via Helm.

## Problem
During installation, it is not possible to specify the framework required for a specific cloud provider. As a result, I must manually extract the necessary controls from the JSON reports, which can be a time-consuming process.

## Solution
Introduce a capability similar to Trivy Operator, allowing users to specify one or multiple frameworks in the `values.yaml` file. For example:

```bash
  specs:  
    - k8s-cis-1.23  
    - k8s-nsa-1.0  
    - k8s-pss-baseline-0.1  
    - k8s-pss-restricted-0.1  

Additionally, enable this configuration through the `--set` flag in the Helm install command.

## Additional Information
I would want to be able to set for GKE:
```bash
frameworks:
- AllControls
- cis-v1.23-t1.0.1

For AKS:
```bash
framworks:
- cis-aks-t1.2.0

**Comment by dwertent** [comment]
1. [ ] [Add the config](https://github.com/armosec/armoapi-go/pull/421) 
2. [ ] Update the [dependencies](https://github.com/armosec/utils-k8s-go/blob/master/go.mod#L6) 
3. [ ] Add the code to the Operator and release the Operator 
4. [ ] Add the default values to the Helm and release the Helm

**Comment by slashben** [comment]
@entlein will take this issue

**Comment by manumathon** [comment]
@matthyx should I take this up.

**Comment by matthyx** [comment]
@manumathon yes, please go ahead — this is still open and unclaimed. Some context so you don't have to rediscover it, since parts of it have moved since the issue was filed.

### What already landed

Step 1 of the plan in https://github.com/kubescape/helm-charts/issues/566#issuecomment-2548761633 is done: armosec/armoapi-go#421 merged 2025-04-04 and `DefaultFrameworks []string` now exists on `InstallationData` in `armotypes/portaltypes.go`. Nothing after that was implemented — there's no `DefaultFrameworks` reference in `utils-k8s-go` or in `kubescape/operator`, and no `defaultFrameworks` key in the chart (currently 1.40.3).

So the remaining scope is steps 2–4:

- [x] Add the config — armosec/armoapi-go#421
- [ ] Update the [dependencies](https://github.com/armosec/utils-k8s-go/blob/master/go.mod) to pick up the new field
- [ ] Consume `defaultFrameworks` in the Operator and release it
- [ ] Expose `defaultFrameworks` in the chart's `values.yaml` and release it

### Existing partial workaround

Worth knowing before you start: the chart gained a partial escape hatch after this issue was opened. `kubescapeScheduler.requestBody.commands[].args.scanV1` already accepts a framework selector at install time:

```yaml
kubescapeScheduler:
  requestBody:
    commands:
      - CommandName: "kubescapeScan"
        args:
          scanV1: {"targetType": "framework", "targetNames": ["nsa"]}
```

That covers the original ask (scan only `cis-aks-t1.2.0` rather than everything) **for the scheduled CronJob scan only**. It is not a substitute for this issue, because:

1. it applies to the scheduler CronJob, not as a cluster-wide default for every scan;
2. it's a raw JSON blob rather than the list form + `--set` ergonomics requested here;
3. the override semantics from https://github.com/kubescape/helm-charts/issues/566#issuecomment-2548716838 (API-triggered frameworks take precedence over the installation defaults) were never implemented.

Also worth fixing along the way: the comment above `kubescapeScheduler` in `values.yaml` still says *"By default it will scan all resources in the cluster for all frameworks"*, which will be inaccurate once `defaultFrameworks` exists.

### Suggested naming

Please stick with `defaultFrameworks` (not `frameworks`) to match the field already merged in `armoapi-go` and to make the "API call overrides the install-time default" relationship read correctly.

**Comment by manumathon** [comment]
@matthyx ready for review — remaining steps from your earlier notes:

* Operator: https://github.com/kubescape/operator/pull/398
* Helm chart: https://github.com/kubescape/helm-charts/pull/890
* Docs: https://github.com/kubescape/kubescape.io/pull/135

Operator first (reads `defaultFrameworks`); chart writes the value; docs cover install/`--set` usage.
